common/Util.cpp

```diff
@@ -125,16 +130,23 @@ namespace Util
 
             // a poor fallback but something.
             std::vector<char> random = getBytes(length);
-            int fd = open("/dev/urandom", O_RDONLY);
             int len = 0;
+#ifdef HAVE_SYS_RANDOM_H
+            len = getrandom(random.data(), length, GRND_NONBLOCK);
+            if(len < 0)
```

Comment:
FYI, we use clang-format to format code (there .clang-format file in the project). (There is always a space after `if`, `for`, and `while`.)